mapswipe_workers/python_scripts/remove_duplicated_information_in_postgres.py

```diff
@@ -0,0 +1,130 @@
+    """Get project ids."""
+
+    p_con = auth.postgresDB()
+
+    query = """
+        SELECT project_id FROM projects
+        LIMIT 10
+    """
+    data = p_con.retr_query(query)
+    project_ids = []
+    for item in data:
+        project_ids.append(item[0])
+
+    logger.info("Got projects from postgres.")
+    logger.info(project_ids)
+    return project_ids
+
+
+def update_tasks_table(project_id: str):
+    """Remove duplicates in 'project_types_specifics' attribute in tasks table."""
+
+    logger.info(f"Start process for project: '{project_id}'")
+    p_con = auth.postgresDB()
+
+    query = """
+            UPDATE tasks
+            SET project_type_specifics = project_type_specifics::jsonb
+                #- '{projectId}'
+                #- '{groupId}'
+                #- '{taskId}'
+                #- '{geometry}'
+                #- '{wkt}'
+            WHERE project_id = %(project_id)s
+        """
+    try:
+        p_con.query(query, {"project_id": project_id})
+        logger.info(f"Updated tasks table for project '{project_id}'.")
+    except Exception as e:
+        sentry.capture_exception(e)
+        sentry.capture_message(
+            f"Could NOT update tasks table for project '{project_id}'."
+        )
+        logger.exception(e)
+        logger.warning(f"Could NOT update tasks table for project '{project_id}'.")
+
+
+def update_groups_table(project_id: str):
+    """Remove duplicates in 'project_types_specifics' attribute in groups table."""
+
+    logger.info(f"Start process for project: '{project_id}'")
+    p_con = auth.postgresDB()
+
+    # removing only the duplicated keys would be better
+    """
+    select
+        project_type_specifics::jsonb #- '{projectId}' #- '{groupId}'
+    from tasks
+    limit 10
+    """
+
+    query = """
+        UPDATE groups
+        SET project_type_specifics = project_type_specifics::jsonb
+            #- '{projectId}'
+            #- '{id}'
+            #- '{requiredCount}'
+            #- '{finishedCount}'
+            #- '{neededCount}'
+            #- '{reportCount}'
+            #- '{distributedCount}'
+        WHERE project_id = %(project_id)s
+    """
+    try:
+        p_con.query(query, {"project_id": project_id})
+        logger.info(f"Updated tasks table for project '{project_id}'.")
+    except Exception as e:
+        sentry.capture_exception(e)
+        sentry.capture_message(
+            f"Could NOT update tasks table for project '{project_id}'."
+        )
+        logger.exception(e)
+        logger.warning(f"Could NOT update tasks table for project '{project_id}'.")
+
+
+def run_vacuum_tasks_table():
+    """Run vacuum to reclaim storage."""
+    logger.info("Start vacuum on tasks table.")
+    p_con = auth.postgresDB()
+    # isolation_level 0 will move you out of a transaction block
+    old_isolation_level = p_con._db_connection.isolation_level
+    p_con._db_connection.set_isolation_level(0)
+    query = """
+            VACUUM FULL tasks
```

Comment:
I'm not 100% sure what the schema for the tables are, but what happens at the end of the entire operation? Do you remove some columns from the table? If so, is the vacuuming even needed?
From the docs, `VACUUM FULL` is slow and locks the table for the entire duration of operation. Is this potentially problematic? It also does a full copy of the table, so you'd need to have enough spare disk space to hold that.

Reply:
Hey @laurentS ,
So this is also a very good comment and I didn't really think about it.

I was first confused that after I ran the script on a table in dev, that the actual size of the table increased, although I removed information. Then I found out that you have to run `vacuum` to reclaim the storage.

But what you say makes totally sense and I also read a bit further about this. So it seems the only solution is to make sure to have enough disk space. In our case this would mean that we would need an additional 50GB to perform this operation.

Not sure what is the best forward from here now...